.dev/compare_branches.R

```diff
@@ -311,15 +313,23 @@ lint_one_package <- function(package, linters, out_dir, check_deps) {
   #   ignore them because they're innocuous.
   # also ignore lintr's deprecations to make including all linters easier
   suppressMessages(withCallingHandlers(
-    lints <- as.data.frame(lint_dir(package, linters = linters, parse_settings = FALSE)),
+    {
+      # Don't fail out if one branch exhibits errors (e.g. if the new version fixes a known error)
+      lints <- tryCatch(lint_dir(package, linters = linters, parse_settings = FALSE), error = identity)
```

Comment:
You are already within a `withCallingHandlers()` block so you could just add an error handler?